lib/twilio-ruby/http/http_client.rb

```diff
@@ -52,6 +50,14 @@ def _request(request)
         twilio_response
       end
 
+      def send(request)
+        @connection.send(request.method.downcase.to_sym,
+                         request.url,
+                         request.method == 'GET' ? request.params : request.data)
+      rescue Faraday::ClientError => e
+        raise Twilio::REST::TwilioError, e
```

Comment:
fwiw, the way I handled this was

```ruby
  rescue Faraday::ConnectionFailed => e
    # NOTE(BF): see https://github.com/twilio/twilio-ruby/issues/455
    message = "#{e.class.name} #{e.message}"
    error = Twilio::REST::TwilioError.exception(message)
    error.set_backtrace e.backtrace
    raise error
```

Reply:
Changing to a _breaking_ change since `Faraday::ConnectionFailed` errors are no longer leaked which may break existing error handling.

Since the original error is contained in the wrapper error `cause` field, I don't think it's necessary to modify the message and backtrace.